src/html-reporter/es6-map.js

```diff
@@ -1,14 +1,35 @@
-// Support IE 9-10, PhantomJS: Fallback for fuzzysort.js used by ./html.js
+// Support IE 9-10, Safari 7, PhantomJS: Partial Map fallback.
+// Used by html.js (via fuzzysort.js), and test.js.
+//
+// FIXME: This check is broken. This file is embedded in the qunit.js closure,
+// thus the Map var is hoisted in that scope, and starts undefined (not a function).
 var Map = typeof Map === "function" ? Map : function StringMap() {
```

Comment:
Should we just fix it now?
```suggestion
if ( typeof Map !== "function" ) {
	var Map = function StringMap() {
```

Reply:
That won't work, it'll still be hosted just the same, right? It's a pretty tough situation. I set it up such that it is included by rollup as `intro` (within the file closure, so as to not leak and be seen by end-user source code), and thus the variable will be seen by fuzzysort.js and other code we use naturally as if it was a global.

Without pulling in `globalThis` into here, I'm not sure how to do this because as soon as you declare anything as `Map` you inherently also deprive your ability to see the outer scope's value for that same reference.

Reply:
Oh, right. What's needed is something like the following, which can't be fixed at this level.
```js
(function(…, Map, …) {
	if ( typeof Map !== "function" ) {
		Map = function StringMap() {…};
	}
})(…, typeof Map === "function" && Map, …)
```